src/dbnode/storage/types.go

```diff
@@ -1285,6 +1296,9 @@ type newFSMergeWithMemFn func(
 	dirtySeriesToWrite map[xtime.UnixNano]*idList,
 ) fs.MergeWith
 
+// NewBackgroundProcessFn is a function that creates and returns a new BackgroundProcess.
```

Comment:
nit: `BackgroundProcessFn` reads better imo; otherwise maybe `BackgroundProcessBuilder`?

Reply:
Not really sure about this. Was trying to be consistent with prior art (and it seems to be reasonable to me).
`git grep 'type New\w*Fn func' | wc -l` yields 18 such types for me.